nx/source/kernel/utimer.c

```diff
@@ -0,0 +1,35 @@
+// Copyright 2018 plutoo
+#include "kernel/svc.h"
+#include "kernel/utimer.h"
+#include "arm/counter.h"
+
+void utimerCreate(UsermodeTimer* t, u64 interval, bool start)
+{
+    t->next_time = 0;
+    t->interval = armNsToTick(interval);
+
+    if (start)
+        utimerStart(t);
+}
+
+void utimerStart(UsermodeTimer* t)
+{
+    __sync_bool_compare_and_swap(&t->next_time, 0, armGetSystemTick() + armNsToTick(t->interval));
```

Comment:
__sync_* functions are deprecated GCC intrinsics. Please replace them with the modern [__atomic_*](https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html) functions. Also, compare-and-swap is an operation that should be in a loop, or else this is broken.